service/health/service.go

```diff
@@ -30,17 +34,114 @@ func New(config Config) (*Service, error) {
 	return s, nil
 }
 
-// Search searches for the cluster information.
-// It try to find cluster information in CR and fallback to storage service when nothing is found.
+// Search accepts data from the control plane and tenant cluster and returns
+// an object representing an aggregate health of the cluster and its nodes.
 func (s *Service) Search(ctx context.Context, request Request) (*Response, error) {
-	clusterHealth := NewClusterStatus(request.Cluster, request.Nodes)
-	nodesHealth := NewNodesStatus(request.Nodes)
-	nodesHealth = FillNodeVersions(nodesHealth, request.Cluster.Nodes)
+	clusterStatus := calculateClusterStatus(request.Cluster, request.Nodes)
+	nodeStatus := calculateNodeStatus(request.Cluster, request.Nodes)
 
 	response := Response{
-		Cluster: clusterHealth,
-		Nodes:   nodesHealth,
+		Cluster: clusterStatus,
+		Nodes:   nodeStatus,
 	}
 
 	return &response, nil
 }
+
+func calculateClusterStatus(cluster v1alpha1.StatusCluster, nodes []v1.Node) ClusterStatus {
```

Comment:
This thing definitely deserves some juicy unit tests.

Reply:
I've added unit tests for `calculateClusterStatus` and `calculateNodeStatus`. There is 100% test coverage now.